### ISSUE TITLE
Cannot pass extra query parameters with functions

### ISSUE THREAD

**Issue by melkamar** [grafana/grafana#6131]
* **I'm submitting a ...**
- [x] Bug report
- [x] Feature request

In Grafana I want to use the summarize() function with month buckets. I want the buckets to start with the first day of the month. The API looks like this:

[`&target=summarize(metric, "13week", "avg", true)&from=midnight+20100101`](http://graphite.readthedocs.io/en/latest/functions.html#graphite.render.functions.summarize)

However, there is no way in Grafana to append the `&from=...` parameter. Therefore I am stuck with the default value which seems to be the 15th or 16th of a month. Appending the parameter in the Edit mode does not do anything.

Using Grafana 3.1.1 with Graphite on Ubuntu.

**Comment by torkelo** [comment]
you can add optional function parameters using by adding `,` as you type the value in the argument 

http://docs.grafana.org/datasources/graphite/#optional-parameters

**Comment by melkamar** [comment]
I don't think we mean the same thing, I have read the Grafana docs. Optional parameters inside the parentheses are fine, but the `&from=` query parameter is not a parameter _inside_ the function. 

![image](https://cloud.githubusercontent.com/assets/3999871/18958592/5cb9df7e-8663-11e6-8059-c95acfbff745.png) 
This function works as intended, _but_ I need to append the `&from=midnight+20100101` query somewhere. Putting it in the parenthesis does not work.

**Comment by torkelo** [comment]
You control the time range via the time picker at the top right of the dashboard menu,  or you can put in a override in time range tab of the panel

**Comment by melkamar** [comment]
I am aware of that function. This, however is something different. This parameter tells Graphite how to _align_ the data, not the range that should be shown.

The doc says: 
> Passing alignToFrom=true will instead create buckets starting at the from time. In this case, the bucket for 22:32 depends on the from time. If from=6:30 then the 1hour bucket for 22:32 is 22:30-23:30.

So using the range picker I tell Grafana what range of dates I want to use, whereas passing this parameter will tell it how do I want the data to be aligned (e.g. showing one datapoint per month - should this datapoint be the first day of month? the middle? end?)

**Comment by torkelo** [comment]
so why not specify alignToFalse=true as the third param?
![image](https://cloud.githubusercontent.com/assets/10999/19144821/f5381d54-8bab-11e6-9636-b9b8ad262d9f.png)

**Comment by torkelo** [comment]
actually that was stupid :) strange that it worked

**Comment by torkelo** [comment]
the from parameter is not part of they query expression but the time range, you specify the time range in the timepicker. you can have the time picker round to even days or weeks or months by for example 
`now-5M/M`

**Comment by melkamar** [comment]
If that is so then the (perceived?) error is at Graphite's side. I wanted data to be summarized per month and have a datapoint on the first day of each month. What I am getting is the first datapoint being on the first day, but then as months pass by the date starts "moving backwards" and the datapoints are on 31st, 30th ... of the previous month. I assumed this was due to a wrong setup of the `summarize` function.

tl;dr Thank you for the patience, I understand now that this is not an error in Grafana.
